### PR TITLE
Placements(subnet) selection UI improvements

### PR DESCRIPTION
**Pull request by tylerwowen**
UI improvements for creating new and updating existing clusters in the advanced mode:
- Placements are grouped by Availability Zone.
- Placements options text is extended to include capacity and abscrac name.
- Placements options text are marked by different colors to reflect remaining capacity.

Code improvements:
- Consolidate all instances that use a repetitive callback function to convert from placements to options. The new implementation only uses 2 functions `getSimpleList` and `getFullList`.
- Simplified `getFullList` implementation.
- Simplified `inDefault` implementation.
- Fixed invalid `this` reference and non-exist `this.isdefault`.

### CODE REVIEW

**Review by liyaqin1 — APPROVED** [review]
If you test ok, then feel free to land it.
If you could put some pictures here for us to know how the improvement look like, it would be better.

**Comment by tylerwowen** [comment]
> If you test ok, then feel free to land it. If you could put some pictures here for us to know how the improvement look like, it would be better.

I had screen shots in the Jira ticket CDP-4776. They were removed from this PR because of security concerns.

File: deploy-board/deploy_board/static/js/deploy-board.js [hunk 1]
```diff
@@ -202,60 +233,31 @@ function getDefaultPlacement(capacityCreationInfo) {
...
-                        isSelected: this.isDefault(item)
+                        isSelected: inDefault(item)
                     }
                 })
             }
-            return arr;
-
+            return convertToPlacementOptions(arr);
         },
         getFullList: function (showPublicOnly, existingItems) {
```
Comment (robbintt) [inline deploy-board/deploy_board/static/js/deploy-board.js:242]: do we have any unit testing framework for js in deploy board?

File: deploy-board/deploy_board/static/js/deploy-board.js [hunk 2]
```diff
@@ -89,6 +91,57 @@ function getDefaultPlacement(capacityCreationInfo) {
...
+        }
+        return options;
+    }
+
+    function convertToPlacementOptions(placements) {
+        return {'Basic Settings': placements};
+    }
+
+    function inDefault(item) {
+        const searchSource = item.assign_public_ip ? topCmpPublicIPPlacements : topCmpPrivateIPPlacements;
```
Comment (robbintt) [inline deploy-board/deploy_board/static/js/deploy-board.js:139]: Does this mean the checkbox decides the default subnets or does this affect search visibility?

Comment (tylerwowen) [reply deploy-board/deploy_board/static/js/deploy-board.js:139]: `assign_public_ip` filed of a placement object is stored in the DB and indicates whether the placement is a private or a public subnet. It's different from the UI button "Assign Public IP".

File: deploy-board/deploy_board/static/js/deploy-board.js (hunk 1 above)
Comment (tylerwowen) [reply deploy-board/deploy_board/static/js/deploy-board.js:242]: No, we don't.

**Review by robbintt — DISMISSED** [review]

**Review by robbintt — APPROVED** [review]

**Pull request merged**